toolkit/smart-contracts/offchain/src/await_tx.rs

```diff
@@ -4,40 +4,46 @@ use sidechain_domain::UtxoId;
 use std::time::Duration;
 use tokio_retry::{Retry, strategy::FixedInterval};
 
+/// Trait for [await_tx_output] to allow different awaiting strategies.
 pub trait AwaitTx {
 	#[allow(async_fn_in_trait)]
-	async fn await_tx_output<Q: QueryUtxoByUtxoId>(
+	/// Keeps querying `utxo_id` transaction output through `client` with some strategy depending on the implementation.
+	/// This is used for blocking until a submitted transaction is done.
+	async fn await_tx_output<C: QueryUtxoByUtxoId>(
 		&self,
-		query: &Q,
+		client: &C,
 		utxo_id: UtxoId,
 	) -> anyhow::Result<()>;
 }
 
+/// Type representing retries of a maximum amount and with a fixed delay.
```

Comment:
```suggestion
/// Type representing retries of a number amount and with a fixed delay.
```
https://www.grammarly.com/commonly-confused-words/amount-vs-number